helix-rest/src/main/java/org/apache/helix/rest/client/CustomRestClientImpl.java

```diff
@@ -122,18 +146,23 @@ protected JsonNode getJsonObject(HttpResponse httpResponse) throws IOException {
 
   @VisibleForTesting
   protected HttpResponse post(String url, Map<String, String> payloads) throws IOException {
+    HttpPost postRequest = new HttpPost(url);
+    postRequest.setHeader("Accept", ACCEPT_CONTENT_TYPE);
+    StringEntity entity = new StringEntity(OBJECT_MAPPER.writeValueAsString(payloads), ContentType.APPLICATION_JSON);
+    postRequest.setEntity(entity);
+    HttpResponse httpResponse;
+    int hashCode = url.hashCode() + payloads.hashCode();
```

Comment:
url1 has hash = 1 and payloads1 has hash = 3.
url2 has hash = 2 and payloads2 has hash = 2.
Their merged hashCode will be the same with this code.

Reply:
Yes, the hashCode doesn't prevent the trick conditions. The implementation is based on our prior knowledge about the incoming parameter (baseUrl + custom) and it won't cause any conflictions. Another reason is the `HttpRequest` class in Java library, it doesn't handle `equals` and `hashCode` function very well, the same baseUrl ends up different by the computation.